internal/domain/migration/yaml.go

```diff
@@ -39,38 +37,52 @@ func ReadDir(dir string, set *Set) error {
 	if err != nil {
 		return errors.Wrap(err, "failed to read directory")
 	}
+
 	for _, f := range files {
 		if f.IsDir() {
 			if err := ReadDir(filepath.Join(dir, f.Name()), set); err != nil {
 				return err
 			}
 			continue
 		}
-		if filepath.Ext(f.Name()) != ".yml" && filepath.Ext(f.Name()) != ".yaml" {
+
+		if filepath.Ext(f.Name()) != ".sql" {
 			continue
 		}
-		priority := 1
+		servicePriority := 1
+		serviceName := ""
+		migrationPriority := 0
+
 		if parts := strings.Split(f.Name(), "_"); len(parts) > 1 {
 			if p, err := strconv.Atoi(parts[0]); err == nil {
-				priority = p
+				migrationPriority = p
+			}
+		}
+
+		if folders := strings.Split(dir, "/"); len(folders) > 1 {
```

Comment:
Could you explain what this code does?

Reply:
@darland funny enough this code had a bug! Probably you can just smell them.
Tried to explain here https://github.com/webdevelop-pro/migration-service/blob/feat/example/internal/domain/migration/sql.go#L66